plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveMetastore.java

```diff
@@ -51,6 +51,11 @@
 
     void updatePartitionStatistics(HiveIdentity identity, Table table, String partitionName, Function<PartitionStatistics, PartitionStatistics> update);
 
+    default void updatePartitionStatistics(HiveIdentity identity, Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        updates.forEach((partitionName, update) -> updatePartitionStatistics(identity, table, partitionName, update));
+    }
+
```

Comment:
Alternatively you can define `updatePartitionStatistics(HiveIdentity identity, Table table, String partitionName, Function<PartitionStatistics, PartitionStatistics> update);` as a default method redirecting to override which takes a map, and pass a singleton map there.

Then you can drop overrides for `void updatePartitionStatistics(HiveIdentity identity, Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)` in `CachingHM` and `RecordingHM`, and instead implement the method which takes a map in leaf implementations (`Bridging`, `File`, `Alluxio`).

Both approaches are fine I think, though.